upload/src/addons/SV/WarningImprovements/XF/Entity/Warning.php

```diff
@@ -19,9 +19,10 @@ public function canView(&$error = null)
             return false;
         }
 
-        if ($visitor->user_id === $this->user_id)
+        if ($visitor->user_id === $this->user_id && $this->app()->options()->sv_view_own_warnings)
         {
-            return $this->app()->options()->sv_view_own_warnings;
+            $handler = $this->getHandler();
+            return $handler ? $handler->canViewContent($this) : false;
```

Comment:
This prevents a user from seeing a warning for content which has been deleted or is inaccessible. A `return true` works.